src/formio/validators/minMaxDateAndDatetimeValidator.js

```diff
@@ -0,0 +1,48 @@
+import set from 'lodash/set';
+
+import {validateBoundaries} from './utils';
+
+const createValidator = (key, invalidKey) => {
+  return {
+    key,
+    message(component) {
+      const minDate = new Date(component.component.minDate);
+      const maxDate = new Date(component.component.maxDate);
+
+      const errorKey =
+        component.openForms.validationErrorContext.minMaxDateAndDatetimeValidatorErrorKey;
+      const componentErrorMessages = component.component.errors;
```

Comment:
I'm confused, we want to show the most appropriate error message, so `errorKey` is for example `minDate` - why can't we use `component.errorMessage(errorKey)`?

e.g. with `minDate`, this will:

1. Look up the key in the custom error messages and use it if it's present
2. Look up the global message/translation if it's not present in the custom keys

If we don't have a specific error key, the error key itself should become `invalid_date` or `invalid_datetime`, which can in turn again be looked up with the custom error messages or the global translation.

We always want to use the most specific error keys, so `minDate` global translation is favored over `invalid_date` custom error message. I think I misunderstood your [comment](https://github.com/open-formulieren/open-forms-sdk/pull/671#discussion_r1567504978)

Reply:
The issue with `component.errorMessage(errorKey)` is that it will fallback to `errorKey`:

```js
  errorMessage(type) {
    return (this.component.errors && this.component.errors[type]) ? this.component.errors[type] :  type;
  }
```

So instead of doing this currently:

```js
const errorMessage = componentErrorMessages[errorKey] || componentErrorMessages[invalidKey] || errorKey;
```

I'd have to do:

```js
const errorMessage = component.errorMessage(errorKey) === errorKey ? component.errorMessage(invalidKey) === invalidKey ? errorKey : invalidKey;
```
 (not sure if this is a 100% correct), but it is hard to understand

Reply:
but the fallback to `errorKey` is fine, because then it'll use the translation from https://github.com/open-formulieren/open-forms/blob/master/src/openforms/js/lang/formio/nl.json

Reply:
> We always want to use the most specific error keys, so `minDate` global translation is favored over `invalid_date` custom error message. I think I misunderstood your [comment](https://github.com/open-formulieren/open-forms-sdk/pull/671#discussion_r1567504978)

Reading this again, I'm not sure I understand what would be the use case for `invalid_date` in this case? If we fallback to the `minDate` global translation, `invalid_date` will never be used?

It might be that you want the `invalid_date` error key to be used when the input is not a valid date (e.g. `2020-ab-cd`), but you can't input non-numeric characters in the SDK, and when giving an "incomplete" date (e.g. `2020-__-__`), formio will autocomplete the rest.

Reply:
`invalid_date` is used when the input cannot be parsed as a date: https://github.com/formio/formio.js/blob/bebc2ad73cad138a6de0a8247df47f0085a314cc/src/validator/Validator.js#L526

Reply:
:+1: 
The time component would also have to follow this path as well (in another PR)

Reply:
I'll note that trying to set the date to an invalid input via backend logic (and I assume via other ways too) crashes the SDK, before [this validator](https://github.com/formio/formio.js/blob/bebc2ad73cad138a6de0a8247df47f0085a314cc/src/validator/Validator.js#L526) is being called, so the `invalid_date` key is currently not being used

Reply:
hrm, concerning, but let's just stick to the formio patterns here. There were some flags with allowing/not allowing invalid dates so this is probably relevant for old components still.